packages/web-components/fast-foundation/src/custom-properties/manager.ts

```diff
@@ -0,0 +1,282 @@
+     */
+    protected _owner: CustomPropertyManagerClient | null = null;
+
+    /**
+     * Tracks state of rAF to only invoke property writes once per animation frame
+     */
+    private ticking: boolean = false;
+
+    /**
+     * Stores all CSSCustomPropertyDefinitions registered with the provider.
+     */
+    private cssCustomPropertyDefinitions: Map<
+        string,
+        CSSCustomPropertyDefinition & { count: number }
+    > = new Map();
+
+    /**
+     * {@inheritdoc CustomPropertyManager.register}
+     */
+    public register(def: CSSCustomPropertyDefinition): void {
+        const cached = this.cssCustomPropertyDefinitions.get(def.name);
+
+        if (cached) {
+            cached.count += 1;
+        } else {
+            this.cssCustomPropertyDefinitions.set(def.name, {
+                ...def,
+                count: 1,
+            });
+            this.set(def);
+        }
+    }
+
+    /**
+     * {@inheritdoc CustomPropertyManager.unregister}
+     */
+    public unregister(name: string): void {
+        const cached = this.cssCustomPropertyDefinitions.get(name);
+
+        if (cached) {
+            cached.count -= 1;
+
+            if (cached.count === 0) {
+                this.cssCustomPropertyDefinitions.delete(name);
+                this.deleteCustomProperty(name);
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc CustomPropertyManager.set}
+     */
+    public set = (definition: CSSCustomPropertyDefinition) => {
+        if (this.owner) {
+            this.customPropertyTarget.setProperty(
+                `--${definition.name}`,
+                this.owner.evaluate(definition)
+            );
+        }
+    };
+
+    /**
+     * {@inheritdoc CustomPropertyManager.set}
+     */
+    public setAll() {
+        if (this.ticking) {
+            return;
+        }
+
+        this.ticking = true;
+        DOM.queueUpdate(() => {
+            this.ticking = false;
+            this.cssCustomPropertyDefinitions.forEach(def => {
+                this.set(def);
+            });
+        });
+    }
+
+    /**
+     * Removes a CSS custom property from the provider.
+     */
+    private deleteCustomProperty = (name: string): void => {
+        this.customPropertyTarget.removeProperty(`--${name}`);
+    };
+}
+
+/**
+ * An implementation of {@link CustomPropertyManager} that uses the constructable CSSStyleSheet object.
+ * This implementation supports multiple CustomPropertyManagerTarget subscriptions.
+ *
+ * @public
+ */
+export class ConstructableStylesCustomPropertyManager extends CustomPropertyManagerBase {
+    protected readonly sheet: CSSStyleSheet;
+    protected styles: ElementStyles;
+    protected customPropertyTarget: CSSStyleDeclaration;
+    private subscribers = new Set();
+
+    constructor(sheet: CSSStyleSheet) {
+        super();
+
+        this.sheet = sheet;
+
+        this.styles = ElementStyles.create([sheet]);
+
+        this.customPropertyTarget = (sheet.rules[
```

Comment:
Should we use `cssRules` since `rules` is considered legacy? https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/rules

Reply:
Good catch, updated.